core/store/orm/orm.go

```diff
@@ -326,6 +339,44 @@ func (orm *ORM) LinkEarnedFor(spec *models.JobSpec) (*assets.Link, error) {
 	return earned, nil
 }
 
+// UpsertErrorFor upserts a JobSpecError record, incrementing the occurences counter by 1
+// if the record is found
+func (orm *ORM) UpsertErrorFor(jobID *models.ID, description string) {
+	orm.MustEnsureAdvisoryLock()
+	err := orm.db.Exec(`
+		INSERT INTO job_spec_errors (
```

Comment:
This is cool. You can also use default gorm with this as well, grep for `.Set("ON CONFLICT"`

Reply:
hmmm I feel like this is better because now we can let gorm auto set the created_at and updated_at fields..